src/Zap.sol

```diff
@@ -386,83 +428,108 @@ contract Zap is Errors {
         bytes calldata _params
     )
         external
+        onlyAave
+        requireStage(Stage.LEVEL1)
         returns (bool)
     {
-        (
-            uint128 _accountId,
-            uint128 _collateralId,
-            uint256 _zapTolerance,
-            uint256 _swapTolerance,
-            address _receiver
-        ) = abi.decode(_params, (uint128, uint128, uint256, uint256, address));
+        stage = Stage.LEVEL2;
 
-        (uint256 unwound, address collateral) = _unwind(
-            _flashloan,
-            _premium,
-            _accountId,
-            _collateralId,
-            _zapTolerance,
-            _swapTolerance
+        (,,,,,, address _receiver) = abi.decode(
+            _params,
+            (uint128, uint128, uint256, uint256, uint256, uint256, address)
         );
 
-        _flashloan += _premium;
+        (uint256 unwound, address collateral) =
+            _unwind(_flashloan, _premium, _params);
+
```

Comment:
is it worth emitting an event here for easier tracking of values / improved transparency?

(e.g. `Unwound(...)`)